rust-server/src/conn.rs

```diff
@@ -59,10 +73,104 @@ impl dyn Connection {
                 let events = JsonFileWriter::new("events".to_owned()).context("opening events")?;
                 JsonFiles { streams, events }
             }),
+            Storage::Postgres => Box::new({
```

Comment:
I think you could have a separate opener for the Postgres storage so the arguments that aren't relevant don't get passed.